lib/model/model.go

```diff
@@ -1242,65 +1241,55 @@ func (m *Model) ConnectedTo(deviceID protocol.DeviceID) bool {
 }
 
 func (m *Model) GetIgnores(folder string) ([]string, []string, error) {
-	var lines []string
-
 	m.fmut.RLock()
 	cfg, ok := m.folderCfgs[folder]
 	m.fmut.RUnlock()
 	if !ok {
-		return lines, nil, fmt.Errorf("Folder %s does not exist", folder)
-	}
-
-	if !cfg.HasMarker() {
-		return lines, nil, fmt.Errorf("Folder %s stopped", folder)
-	}
-
-	fd, err := os.Open(filepath.Join(cfg.Path(), ".stignore"))
-	if err != nil {
-		if os.IsNotExist(err) {
-			return lines, nil, nil
+		cfg, ok = m.cfg.Folders()[folder]
+		if ok {
+			matcher := ignore.New(false)
+			path := filepath.Join(cfg.Path(), ".stignore")
+			if err := matcher.Load(path); err != nil {
+				return nil, nil, err
+			}
+			return matcher.Lines(), matcher.Patterns(), nil
 		}
-		l.Warnln("Loading .stignore:", err)
-		return lines, nil, err
+		return nil, nil, fmt.Errorf("Folder %s does not exist", folder)
 	}
-	defer fd.Close()
 
-	scanner := bufio.NewScanner(fd)
-	for scanner.Scan() {
-		lines = append(lines, strings.TrimSpace(scanner.Text()))
+	if !cfg.HasMarker() {
+		return nil, nil, fmt.Errorf("Folder %s stopped", folder)
 	}
 
 	m.fmut.RLock()
+	lines := m.folderIgnores[folder].Lines()
 	patterns := m.folderIgnores[folder].Patterns()
 	m.fmut.RUnlock()
 
 	return lines, patterns, nil
 }
 
 func (m *Model) SetIgnores(folder string, content []string) error {
+	m.fmut.RLock()
 	cfg, ok := m.folderCfgs[folder]
```

Comment:
I'd try to access the runner instead, and send the can to the runner, so that we don't get the config from two different places (which is very weird).

Also if a folder overall does not exist, you end up checking two places, the map, and then the config.
Instead I'd  use the runner and move checking if the runner exists to the very end of the function, and make that deciding factor if we scan or not, otherwise I'd just trust the config to do the right thing.

```
cfg, ok := Folders()[folder]
if !ok {
   blah
}

do stuff

if runner, ok := m.runners[folder]; ok {
   return runner.Scan()
}
return nil
```